src/global-firestore-data/user-semesters.ts

```diff
@@ -23,6 +23,14 @@ export const editSemesters = (
   });
 };
 
+export const editPlans = (updater: (oldPlans: readonly Plan[]) => readonly Plan[]): void => {
+  const plans = updater(store.state.plans);
+  store.commit('setPlans', plans);
+  updateDoc(doc(semestersCollection, store.state.currentFirebaseUser.email), {
+    plans,
+  });
+};
+
```

Comment:
```suggestion
export const editPlans = async (updater: (oldPlans: readonly Plan[]) => readonly Plan[]): void => {
  const plans = updater(store.state.plans);
  store.commit('setPlans', plans);
  await updateDoc(doc(semestersCollection, store.state.currentFirebaseUser.email), {
    plans,
  });
};
```
`updateDoc` returns a promise that should resolve before `editPlans` returns